working/augmentation-libraries/feature-specification.md

```diff
@@ -1040,136 +765,28 @@ extension type A(int b) {
-    [this section](#non-redirecting-generative-constructors) for more
-    information.
-
-*   **Augmenting operators**: When augmenting an operator, `augmented`
-    refers to the augmented operator method, which must be immediately
-    invoked using function call syntax. For example, when augmenting
-    `operator +` you could use `augmented(1)` to call the augmented
-    operator, and when augmenting `operator []=` you would use the
-    `augmented(key, value)` syntax.
-
-    *   Note that `augmented` in such an augmenting operator method body is
-        not an expression by itself, and cannot be used to tear off the
-        augmented operator method. Similar to `super`, it is a syntactic
-        form which can only be used in limited ways.
-
-*   **Augmenting enum values**: When augmenting an enum value, `augmented`
-    has no meaning and is not allowed.
-
-In all relevant cases, if the augmented member is an instance member, it is
-invoked with the same value for `this`.
-
-Assume that the identifier `augmented` occurs in a source location where no
-enclosing declaration is augmenting. In this case, the identifier is taken
-to be a reference to a declaration which is in scope.
-
-*In other words, `augmented` is just a normal identifier when it occurs
-anywhere other than inside an augmenting declaration.*
-
-*Note that, for example, `augmented()` is an invocation of the augmented
-function or method when it occurs in an augmenting function or method
-declaration. (In the latter case, the augmenting method declaration must
-occur inside an augmenting type-introducing declaration, e.g., an
-augmenting class or mixin declaration). This is also true if `augmented()`
-occurs inside a local function declaration inside the body of that function
-or method declaration. We could say that `augmented` is a contextual
-reserved word because it is unable to refer to a declaration in scope when
-it occurs inside an augmenting declaration, it always has the special
-meaning which is associated with augmentations.*
-
-A compile-time error occurs if a declaration with the basename `augmented`
-occurs in a location where any enclosing declaration is augmenting. *This
-error is applicable to all such declarations, e.g., local functions, local
-variables, parameters, and type parameters.*
-
-Consider a non-augmenting member declaration _Dm_ that occurs inside an
-augmenting type declaration _Dt_. A compile-time error occurs if the
-identifier `augmented` occurs in _Dm_.
-
-*For example, inside `augment class C` we could have a declaration like
-`void f() {...augmented()...}`. This is an error because the outer
-`augment` forces the meaning of `augmented` to be about augmentation in the
-entire scope, but the method declaration is introductory and hence there is
-no earlier declaration to augment.*
+If an augmentation has metadata attached to it, these are appended to the
+metadata of the declaration being augmented.
```

Comment:
I'd say:
> An augmenting declaration can have metadata attached to it.
> Interpretation of metadata is not part of the language,
> and is left to the tools which define the annotations and which
> inspect the source to find them.
> _Some metadata may apply to the combined declaration,
> others to the individual augmenting declarations.
> Only the tool which defined the annotation knows what it means._

Then we can say (somewhere non-normative) that `dart:mirrors` will only work in the combined declarations, and should see the combined annotations, if it works at all.

Reply:
Good catch. I don't particularly care about mirrors and suspect most users don't either. But I wrote a bit of non-normative stuff about the analyzer package.